installers/conda/osx/create_bundle.sh

```diff
@@ -0,0 +1,271 @@
+  local key=$2
+  local string_value=$3
+  /usr/libexec/PlistBuddy -c "Add :$key string $string_value" "$filepath"
+}
+
+function create_disk_image() {
+  local version_name=$1
+  echo "Creating disk image '$version_name.dmg'"
+  # Add additional DragNDrop facilities
+  #   Add link to Applications
+  #   Pad out the image so that we can make changes went mounted
+  #   Store a background image
+  ln -s /Applications "$BUILD_DIR"/Applications
+  mkdir "$BUILD_DIR/.background"
+  cp "$HERE/dmg_background.png" "$BUILD_DIR/.background/background.png"
+  padding_filename=".dummy-padding-file"
+  mkfile 1m "$BUILD_DIR/$padding_filename"
+
+  # Set Finder background
+  hdiutil create -ov -srcfolder "$BUILD_DIR" -volname "$version_name" -fs HFS+ -format UDRW "$TEMP_IMAGE"
+  volume_name=$(hdiutil attach "$TEMP_IMAGE" | perl -n -e '/\/Volumes\/(.+)/ && print $1')
+  trap "cleanup $volume_name" RETURN
+  volume_path=/Volumes/"$volume_name"
+  rm "$volume_path"/"$padding_filename"
+  sed -e "s/@BUNDLE_NAME@/$bundle_name/" "$HERE/dmg_setup.scpt.in" > "$TEMP_OSA_SCPT"
+  osascript "$TEMP_OSA_SCPT" "$version_name"
+  ensure_volume_detached "$volume_name"
+
+  # Create final compressed dmg
+  test -f "$version_name".dmg && rm -f "$version_name".dmg
+  hdiutil convert "$TEMP_IMAGE" -format UDZO -imagekey zlib-level=9 -o "$version_name".dmg
+}
+
+# Print usage and exit
+function usage() {
+  local exitcode=$1
+  echo "Usage: $0 [options] bundle_name icon_file"
+  echo
+  echo "Create a DragNDrop bundle out of a Conda package. The package is built in $BUILD_DIR."
+  echo "This directory will be created if it does not exist or purged if it already exists."
+  echo "The final .dmg will be created in the current working directory."
+  echo "Options:"
+  echo "  -c Optional Conda channel overriding the default mantid"
+  echo
+  echo "Positional Arguments"
+  echo "  bundle_name: The name of the bundle app directory, i.e. the final name will be '${bundle_name}.app'"
+  echo "  icon_file: An icon file, in icns format, for the final bundle"
+  exit $exitcode
+}
+
+## Script begin
+# Optional arguments
+conda_channel=mantid
+while getopts ":c:h" o; do
+  case "$o" in
+  c) conda_channel="$OPTARG";;
+  h) usage 0;;
+  *) usage 1;;
+esac
+done
+shift $((OPTIND-1))
+
+# Positional arguments
+# 2) - the name of the bundle
+# 3) - path to .icns image used as icon for bundle
+bundle_name=$1
+bundle_icon=$2
+
+# Sanity check arguments. Especially ensure that paths are not empty as we are removing
+# items and we don't want to accidentally clean out system paths
+test -n "$bundle_name" || usage 1
+test -n "$bundle_icon" || usage 1
+
+bundle_dirname="$bundle_name".app
+bundle_contents="$BUILD_DIR"/"$bundle_dirname"/Contents
+echo "Building '$bundle_dirname' in '$BUILD_DIR' from '$conda_channel' Conda channel"
+
+# Build directory needs to be empty before we start
+if [ -d "$BUILD_DIR" ]; then
+  echo "$BUILD_DIR exists. Removing before commencing build."
+  rm -fr "$BUILD_DIR"
```

Comment:
This code path will fail for some reason:
```
rm: _bundle_build/MantidWorkbenchNightly.app/Contents/Resources/include: Permission denied
rm: _bundle_build/MantidWorkbenchNightly.app/Contents/Resources/lib/python3.8/site-packages: Permission denied
rm: _bundle_build/MantidWorkbenchNightly.app/Contents/Resources/lib/python3.8: Permission denied
rm: _bundle_build/MantidWorkbenchNightly.app/Contents/Resources/lib: Permission denied
```
...etc.

Despite being able to rm these directories manually?

Reply:
Hmm. I'm not quite sure why this would happen. Was this on a second run of the script or the first time through? What are the permissions on the directory?

Reply:
Second time, I `Ctrl+C`'d midway then restarted, the perms were
`user:wheel` (where user was my account) so I'm not sure why the script struggled ... maybe we leave this for if it comes up again